internal/host/statemachine.go

```diff
@@ -507,7 +507,7 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 	var requiredInputFieldsExist = stateswitch.And(If(IsMachineCidrDefined))
 
 	var isSufficientForInstall = stateswitch.And(If(HasMemoryForRole), If(HasCPUCoresForRole), If(BelongsToMachineCidr), If(IsHostnameUnique), If(IsHostnameValid), If(IsAPIVipConnected), If(BelongsToMajorityGroup),
-		If(AreOcsRequirementsSatisfied), If(AreLsoRequirementsSatisfied), If(AreCnvRequirementsSatisfied), If(SufficientOrUnknownInstallationDiskSpeed), If(SucessfullOrUnknownContainerImagesAvailability), If(HasSufficientNetworkLatencyRequirementForRole), If(HasSufficientPacketLossRequirementForRole), If(HasDefaultRoute))
+		If(AreOcsRequirementsSatisfied), If(AreLsoRequirementsSatisfied), If(AreCnvRequirementsSatisfied), If(SufficientOrUnknownInstallationDiskSpeed), If(SucessfullOrUnknownContainerImagesAvailability), If(HasSufficientNetworkLatencyRequirementForRole), If(HasSufficientPacketLossRequirementForRole), If(HasDefaultRoute), If(IsAPIDomainNameResolvedCorrectly), If(IsAPIInternalDomainNameResolvedCorrectly), If(IsAppsDomainNameResolvedCorrectly))
```

Comment:
Are all these required in order to start installation?

Reply:
Isn't this the entire point of this PR? This is what we agreed on in Slack - we need the 3 of those for installation

Reply:
As Omer mention, all of them are required, each of these validation verifies the resolution of the 3 domains that were discussed.

Reply:
I am not sure they are always available before installation.  Was checked with test-infra cluster ?

Reply:
BM and none platforms ?

Reply:
I have the same question as @ori-amizur, why we need it for anything rather NONE platform?

Reply:
It is alrrady only in none platform, the validation is skipped, see the validation function

Reply:
for example IsAPIDomainNameResolvedCorrectly is needed only for None platform with olm operators

Reply:
@bkhizgiy do this validation apply to SNO, as SNO is running with user-managed-network=true but it doesn't requires setting external dns